common/common-javassist/src/test/java/io/servicecomb/common/javassist/TestJavassistUtils.java

```diff
@@ -54,29 +54,23 @@ public void testField() throws Exception {
 
     Class<?> cls = JavassistUtils.createClass(classConfig);
 
-    // intField
     Field field = cls.getField("intField");
     Assert.assertEquals(Integer.class, field.getType());
 
     Method method = cls.getMethod("getIntField");
     Assert.assertEquals(Integer.class, method.getReturnType());
-    method = cls.getMethod("setIntField", Integer.class);
 
-    // intArrayField
     field = cls.getField("intArrayField");
     Assert.assertEquals(int[].class, field.getType());
 
     method = cls.getMethod("getIntArrayField");
     Assert.assertEquals(int[].class, method.getReturnType());
-    method = cls.getMethod("setIntArrayField", int[].class);
 
-    // listStringField
     field = cls.getField("listStringField");
     Assert.assertEquals("java.util.List<java.lang.String>", field.getGenericType().getTypeName());
 
     method = cls.getMethod("getListStringField");
     Assert.assertEquals("java.util.List<java.lang.String>", method.getGenericReturnType().getTypeName());
-    method = cls.getMethod("setListStringField", List.class);
```

Comment:
deleted "getMethod" is to ensure setter is exist and has correct signature
if has any problems, will throw exception.

Reply:
We need to use assert the verify the call result.

Reply:
If "getMethod" has logic as you mentioned, I think it should be verified rather than just let it be a caller. For example, It is better if we can add the following asset:

method = cls.getMethod("setListStringField", List.class);
assertNotNull(cls.getMethod("setListStringField", List.class))

I will do that in another PR

Reply:
@wldandan Please add a new commit for it.

Reply:
infact, if cls.getMethod return null, will throw exception inside getMethod
so the assert is invalid.

Reply:
if you care this so much, you can invoke org.springframework.util.ReflectionUtils.findMethod(Class<?>, String, Class<?>...)
then if method not exists, will get a null result
but this method invoked getDeclaredMethod inside it, not invoked getMethod, so after assert null, you must assert if it's public.

Reply:
Ok. my thought is straightforward. 
If some methods are called, it should be verified about the effects. That is the purpose of test case. 
As you mentioned, "deleted "getMethod" is to ensure setter is exist and has correct signature if has any problems, will throw exception."

Back the discussion, what the best way you think we should do to verify if do call the following method, could you please suggest? 

method = cls.getMethod("setListStringField", List.class);

Reply:
We still need to verify the method, if the getMethod throw the exception JUnit can still know about and tell us something is wrong.  As a client we want to make sure is the method is not null and we can still use it.   If we want to verify the non-exist method, we also need to add expected exception into JUnit test case to make sure the exception is thrown.

Reply:
@WillemJiang +1. To keep the scope limited, I think we can add another test case to verify "getmethod" for the scenario about throwing exception.
I will do that soon. 
Thanks all guys.